packages/open-truss/src/uqi/iterator.ts

```diff
@@ -0,0 +1,64 @@
+export type IteratorResult<T> = Promise<{
```

Comment:
I know this is probably just implementing the iterator interface, but this file is 🤯. Is there a reason you needed to do this? Are there no standard library utilities we could use / inherit from?

Reply:
> Is there a reason you needed to do this?

I actually stole this from the `trino-client` library (I'll add attribution to the top of the file). It is just a wrapper around  `AsyncIterableIterator` that can be extended to include metadata like the id of a query.

> Are there no standard library utilities we could use / inherit from?

Possible?
- https://github.com/async-generators/iterable was archived in 2019 and doesn't allow extension
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/Iterator looks very similar but is experimental